oop-photoenhancer/PhotoEnhancer/PhotoEnhancer/Filters/ContrastFilter.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PhotoEnhancer
+{
+    class ContrastFilter : PixelFilter
+    {
+       public ContrastFilter(): base (new ContrastParameters()) { }
+
+        public override string ToString()
+        {
+            return "Контраст";
+        }
+
+        public override Pixel ProcessPixel(Pixel originalPixel, IParameters parameters)
+        {
+           
+            var chanelR = Trim((parameters as ContrastParameters).Cofficient * (originalPixel.R - 0.5) + 0.5);
+            var chanelG = Trim((parameters as ContrastParameters).Cofficient * (originalPixel.G - 0.5) + 0.5);
+            var chanelB = Trim((parameters as ContrastParameters).Cofficient * (originalPixel.B - 0.5) + 0.5);
+
+            double Trim(double chanel)
+            {
+                if (chanel > 1)
+                    return 1;
+                else if (chanel < 0)
+                    return 0;
+
+                return chanel;
+            }
+
+            return new Pixel(chanelR, chanelG, chanelB);
+        }
```

Comment:
Все работает, но код страдает дублированием: 
1. вычисление новой интенсивности канала можно вынести в отдельный метод;
2. метод `Trim` уже есть в структуре `Pixel`, зачем его повторять в этом классе?
Вот так будет проще и без дублирования
```suggestion
        public override Pixel ProcessPixel(Pixel originalPixel, IParameters parameters)
        {
            var coeficient = (parameters as ContrastParameters).Cofficient;
            return new Pixel(
                GetContrasty(originalPixel.R, coeficient),
                GetContrasty(originalPixel.G, coeficient),
                GetContrasty(originalPixel.B, coeficient)
                );
        }

        static double GetContrasty(double channel, double coefficient)
        {
            return Pixel.Trim(coefficient * (channel - 0.5) + 0.5);
        }
```
 


 Применение